### PR TITLE
Browser compatibility

### PR DESCRIPTION
**Pull request by Thom1729**
Use `Object.setPrototypeOf` instead of `util.inherits`. With this one change, `node-schedule` should work in browsers (using the standard port of the `events` module).

I'm working on an Electron-based dashboarding application, and I really like `node-schedule`, and this would let me use it.

### CODE REVIEW

**Comment by kibertoad** [comment]
@Thom1729 Could you please add simple airtap-based or Karma-based test for this?

**Comment by Thom1729** [comment]
Do you mean that I should add airtap or Karma to the CI? I haven't used either of those systems, so I might not be the right person to add them. It's possible that I'm misunderstanding.

**Comment by kibertoad** [comment]
@Thom1729 Yup, that's the idea, to test for browser compatibility.
You can use https://github.com/fastify/secure-json-parse/pull/21 as a reference

**Comment by Thom1729** [comment]
Thanks for the clarification; I'll have a go at it.

`Object.setPrototypeOf()` was added in node 0.12.x, so the new 0.10.x test is failing. If 0.10.x compatibility is required, then I can add a fallback, though it wouldn't be a one-line change anymore. (On the browser side, everything five years old and newer supports it, including IE 11. Not sure what your feelings are on stuff older than that. Polyfills are available.)

**Comment by kibertoad** [comment]
@Thom1729 I'm dropping anything below Node 4 for next semver major, which will be after next semver minor, so we could land this one then without changes.

**Comment by kibertoad** [comment]
@Thom1729 Dropped old Node versions, could you please pull latest changes?

**Comment by kibertoad** [comment]
@Thom1729 Btw, I've converted all the tests to Tape, so now it should be really trivial to execute all of them via airtap without doing much changes.

File: lib/schedule.js [hunk 1]
```diff
@@ -165,7 +164,7 @@ function Job(name, job, callback) {
   };
 }
 
-util.inherits(Job, events.EventEmitter);
+Object.setPrototypeOf(Job.prototype, events.EventEmitter.prototype);
```
Comment (kibertoad) [inline lib/schedule.js:167]: Since we are targetting Node 6 now, wouldn't it be better to just use classes and do an `extend`?

**Comment by Thom1729** [comment]
Branch is updated. Working on airtap, but having a bit of an issue with the way that `schedule` is imported in the tests. The following import fails in airtap, but works in tape:

```js
const main = require('../package.json').main;
const schedule = require('../' + main);
```

This one works in both:

```js
const schedule = require('../lib/schedule');
```

I may be missing something obvious, being new to tape and airtap, but just in case I've opened an issue: https://github.com/airtap/airtap/issues/302.

Rewriting the requires to literal strings would fix the issue, but I wouldn't be surprised if there were a better solution.

File: lib/schedule.js (hunk 1 above)
Comment (kibertoad) [reply lib/schedule.js:167]: Actually, no, your solution is better: https://stackoverflow.com/questions/37194202/ie11-gives-script1002-error-when-defining-class-in-javascript

IE 11 is still a thing, so we can't use classes.

Comment (Thom1729) [reply lib/schedule.js:167]: I'd say so. That's probably a change with a larger scope and impact than this one, though.

**Comment by kibertoad** [comment]
@Thom1729 I don't see much value with current way import is done with tests, this will be tested with tsd after we embed types into the repo, so feel free to just use inline paths.

File: lib/schedule.js [hunk 2]
```diff
@@ -6,8 +6,7 @@
   A cron-like and not-cron-like job scheduler for Node.
 */
 
-const events = require('events'),
-  util = require('util'),
+var events = require('events'),
```
Comment (kibertoad) [inline lib/schedule.js:9]: We switched to const

**Review by kibertoad — APPROVED** [review]

**Comment by kibertoad** [comment]
Thanks!

**Comment by kibertoad** [comment]
@Thom1729 How urgently do you need an Electron-compatible version of node-schedule? Wonder if I need to release new version sooner rather than later.

**Comment by Thom1729** [comment]
I'm already using a patched version, so not urgent. Thanks!

**Comment by kibertoad** [comment]
@Thom1729 Gotcha! I'll ping you when 2.0.0 is up then, so that you could return upstream. We are just getting started :D

**Comment by kibertoad** [comment]
@Thom1729 2.0.0 is out now, would appreciate if you could test it out!

**Comment by Thom1729** [comment]
Looks like it's working great! Thanks.

**Pull request merged**